wordpress-mobile/WordPress-Aztec-iOS
# Improves style removal

<h3>Description:</h3>

The changes in this PR improve our style handling mechanisms so that `DOMString` is smarter about inserting new text and applying styles to it.

This PR also changes how styles are applied.  Some styles won't be extended at their edges.

<h3>Performance considerations:</h3>

I haven't measured performance in this PR since improving performance was not one of my goals, but the amount of calls that `DOMString` processes for text insertion and style changes have been reduced dramatically.

The amount of operations executed for inserting text have been reduced greatly as well, since previously we were removing ALL styles for any text insertion, and setting all relevant styles for the range after that.  Now we're just toggling the styles that changed.

<h3>Testing:</h3>

**Test 1:** run all the unit tests.

**Test 2:**

1. Run the editor, insert and remove text, apply and remove styles.
2. Switch to HTML and make sure the HTML is as expected.
3. Switch back to visual and make sure everything's as expected.
